src/test/java/de/retest/recheck/report/TestReportFilterTest.java

```diff
@@ -0,0 +1,208 @@
+						mock( Screenshot.class ), mock( Screenshot.class ), Collections.emptyList() );
+		childDifferences = Arrays.asList( firstChildDifference, secondChildDifference );
+
+		originalElementDifference =
+				new ElementDifference( element, originalAttributesDifference, originalIdentAttributesDifference,
+						mock( Screenshot.class ), mock( Screenshot.class ), childDifferences );
+		when( originalElementDifference.getIdentifyingAttributes() ).thenReturn( identAttributes );
+		when( originalElementDifference.getIdentifyingAttributes().identifier() ).thenReturn( "identifier" );
+		originalRootElementDifference = new RootElementDifference( originalElementDifference, mock( RootElement.class ),
+				mock( RootElement.class ) );
+		secondRootElementDifference = new RootElementDifference( originalElementDifference, mock( RootElement.class ),
+				mock( RootElement.class ) );
+		originalRootElementDifferences = Arrays.asList( originalRootElementDifference, secondRootElementDifference );
+		originalStateDifference =
+				new StateDifference( originalRootElementDifferences, mock( DurationDifference.class ) );
+		originalActionReplayResult = mock( ActionReplayResult.class );
+		when( originalActionReplayResult.getStateDifference() ).thenReturn( originalStateDifference );
+		originalTestReplayResult = new TestReplayResult( "test", 1 );
+		originalTestReplayResult.addAction( originalActionReplayResult );
+		originalSuiteReplayResult =
+				new SuiteReplayResult( mock( ExecutableSuite.class ), 1, mock( GroundState.class ) );
+		originalSuiteReplayResult.addTest( originalTestReplayResult );
+		originalTestReport = new TestReport();
+		originalTestReport.addSuite( originalSuiteReplayResult );
+
+	}
+
+	@Test
+	void Attributes_differences_should_be_filtered_properly() throws Exception {
+		final AttributesDifference filtered =
+				TestReportFilter.filter( mock( Element.class ), originalAttributesDifference, filter );
+		assertThat( filtered.getDifferences() ).containsExactly( notFilterMe );
+	}
+
+	@Test
+	void identifying_attributes_differences_should_be_filtered_properly() throws Exception {
+		when( element.getIdentifyingAttributes() ).thenReturn( mock( IdentifyingAttributes.class ) );
+		final IdentifyingAttributesDifference filtered =
+				TestReportFilter.filter( element, originalIdentAttributesDifference, filter );
+		assertThat( filtered.getAttributeDifferences() ).containsExactly( notFilterMe );
+	}
+
+	@Test
+	void collection_of_element_differences_should_be_filtered_properly() throws Exception {
+		when( element.getIdentifyingAttributes() ).thenReturn( identAttributes );
+		final Collection<ElementDifference> filteredChildDifferences =
+				TestReportFilter.filter( childDifferences, filter );
+		final List<ElementDifference> elementDifferences =
+				filteredChildDifferences.stream().collect( Collectors.toList() );
+		assertThat( elementDifferences.get( 0 ).getAttributesDifference().getDifferences() )
```

Comment:
Maybe split this long expression(s) into one or more variables. (Same elsewhere like line 144.)